### PR TITLE
Move Doctrine constraints out of Other constraints

### PR DESCRIPTION
**Pull request by gitomato**
The "Other Constraints" category is very useful, I regularly find myself checking out each of them to find the best fit. As a user of DBAL only, the Doctrine constraints are not as generic as the other ones.

I suggest giving Doctrine constraints their own category, which may help people that use Doctrine, but also people that don't.

### CODE REVIEW

File: reference/constraints/map.rst.inc [hunk 1]
```diff
@@ -90,6 +90,13 @@ Financial and other Number Constraints
 * :doc:`Issn </reference/constraints/Issn>`
 * :doc:`Isin </reference/constraints/Isin>`
 
+Doctrine Constraints
+~~~~~~~~~~~~~~~~~
```
Comment (xabbuh) [inline reference/constraints/map.rst.inc]:
```suggestion
~~~~~~~~~~~~~~~~~~~~
```

**Review by javiereguiluz — APPROVED** [review]
I think this is a good idea. Thanks @gitomato

**Review by xabbuh — APPROVED** [review]

**Comment by gitomato** [comment]
Well... Not the best timing:

https://symfony.com/blog/new-in-symfony-7-1-expanding-uniqueentity-constraint-to-any-php-class

**Comment by OskarStark** [comment]
Thank you @gitomato.

**Pull request merged**